src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerIndexService.kt

```diff
@@ -53,31 +54,42 @@ class PrisonerIndexService(val nomisService: NomisService,
         return indexStatusService.getCurrentIndex()
     }
 
+    fun addOffendersToBeIndexed(pageRequest : PageRequest) {
+        var count = 0
+        log.debug("Sending offender indexing requests row {} --> {}", pageRequest.offset+1, pageRequest.offset + pageRequest.pageSize)
+        nomisService.getOffendersIds(pageRequest.offset, pageRequest.pageSize).offenderIds?.forEach {
+            indexQueueService.sendIndexRequestMessage(IndexRequest(IndexRequestType.OFFENDER, it.offenderNumber))
+            count += 1
+        }
+        log.debug("Requested {} offender index syncs", count)
+   }
+
     fun indexingComplete() : IndexStatus {
         indexStatusService.markRebuildComplete()
         indexQueueService.clearAllMessages()
         val currentIndex = indexStatusService.getCurrentIndex()
-        log.info("Index marked as Complete, Index {} is now current.", currentIndex.currentIndex)
+        log.info("Index marked as complete, index {} is now current.", currentIndex.currentIndex)
         return currentIndex
     }
 
-    fun addIndexRequestToQueue(): Int {
-        var count = 0
-
-        log.debug("Sending Indexing Requests")
-        var offset = 0
-        do {
-            var numberReturned = 0
-            nomisService.getOffendersIds(offset, pageSizeToRetrieve())?.forEach {
-                indexQueueService.sendIndexRequestMessage(IndexRequest(IndexRequestType.OFFENDER, it.offenderNumber))
-                count += 1
-                numberReturned += 1
-            }
-            offset += pageSizeToRetrieve()
-            log.debug("Requested {} so far, number returned {}", count, numberReturned)
-        } while (numberReturned > pageSize && indexStatusService.getCurrentIndex().inProgress)
-        log.debug("All Rebuild messages sent {}", count)
-        return count
+    fun addIndexRequestToQueue(): Long {
+        log.debug("Sending list of offender requests")
+        var page = 0
+        val totalRows = nomisService.getOffendersIds(0, 1).totalRows
+        if (totalRows > 0) {
+            do {
```

Comment:
So this is now enabling retrieving the blocks of ids in parallel, am I understanding this right?

Reply:
Yes